test/functional/core/server_spec.lua

```diff
@@ -191,9 +192,23 @@ describe('server', function()
     if t.is_os('win') then
       return
     end
-    local peer_addr = n.new_pipename()
-    local client =
-      n.new_session(true, { args = { '--clean', '--listen', peer_addr, '--embed' }, merge = false })
+
+    local peer_temp = n.new_pipename()
+
+    -- Find the last instance of '/' to extract the servername
+    local slash_ind = peer_temp:match('.*/()')
+    if slash_ind == nil then
+      slash_ind = 0
+    end
+    local peer_name = peer_temp:sub(slash_ind)
+    local peer_addr = vim.fs.normalize(fn.stdpath('run') .. '/..') .. '/' .. peer_name
```

Comment:
there are probably better ways but in the worse case this could be
```suggestion
    local peer_addr = ('%s/../%s'):format(vim.fs.normalize(fn.stdpath('run')), peer_name)
```

format() is almost always better. `..` just leads to really clumsy code.